app/models/grda_warehouse/client_file.rb

```diff
@@ -47,39 +49,36 @@ class ClientFile < GrdaWarehouse::File
-
-        sql = sql.or(is_consent_form) if GrdaWarehouse::Config.get(:consent_visible_to_all)
-
-        # Include homeless verification PDfs based on site config:
-        # If using 'release' method, show all files ONLY if there is a valid release. If not, only show your own files.
-        # If using any other method, show all files.
-        if can_view_homeless_verification_pdfs
-          if ::GrdaWarehouse::Config.get(:verified_homeless_history_method).to_sym == :release
-            clients_with_consent = GrdaWarehouse::Hud::Client.active_confirmed_consent_in_cocs(user.coc_codes).select(:id)
-            sql = sql.or(arel_table[:client_id].in(Arel.sql(clients_with_consent.to_sql)).and(is_verified_homeless_history))
-          else
-            sql = sql.or(is_verified_homeless_history)
-          end
+      # show your own files
+      sql = is_own_file
+      # show all verified homeless histories based on site config
+      sql = sql.or(is_verified_homeless_history) if GrdaWarehouse::Config.get(:verified_homeless_history_visible_to_all)
+      # show all consents based on site config
+      sql = sql.or(is_consent_form) if GrdaWarehouse::Config.get(:consent_visible_to_all)
+
+      if user.can_manage_window_client_files?
+        if ::GrdaWarehouse::Config.get(:verified_homeless_history_method).to_sym == :release
+          # show verified homeless histories for clients with full release in the current user's coc
+          #   note: the verified_homeless_history_visible_to_all setting overrides this by including all
+          clients_with_consent = GrdaWarehouse::Hud::Client.active_confirmed_consent_in_cocs(user.coc_codes).select(:id)
+          sql = sql.or(arel_table[:client_id].in(Arel.sql(clients_with_consent.to_sql)).and(is_verified_homeless_history))
+
+          # show all NON-verified-homeless-history files for clients with full releases (regardless of CoC)
+          sql = sql.or(has_full_housing_release.and(is_not_verified_homeless_history))
+        else
+          # show all files for clients with full releases
+          sql = sql.or(has_full_housing_release)
         end
 
         window.where(sql)
       # You can only see files you uploaded
       elsif user.can_see_own_file_uploads? || user.can_use_separated_consent?
-        sql = is_own_file
-        sql = sql.or(is_consent_form) if GrdaWarehouse::Config.get(:consent_visible_to_all)
-        sql = sql.or(is_verified_homeless_history) if can_view_homeless_verification_pdfs
         where(sql)
       # You have specific permission to generate homeless verification PDFs
-      elsif user.can_generate_homeless_verification_pdfs?
+      elsif user.can_generate_homeless_verification_pdfs? && GrdaWarehouse::Config.get(:verified_homeless_history_visible_to_all)
         where(is_verified_homeless_history)
+      elsif user.can_generate_homeless_verification_pdfs?
+        where(is_own_file.and(is_verified_homeless_history))
```

Comment:
I added this after we paired, @eanders. Before, users with `can_generate_homeless_verification_pdfs` could still only see their own files, even if `verified_homeless_history_visible_to_all` was set. I think we agreed that the "visible to all" setting should apply across the board. Let me know if I should change it back.

Reply:
this is perfect.